### PR TITLE
[REEF-581] Add example for restart on YARN

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Adding an example for Driver Restarts in REEF.NET.

JIRA:
  [REEF-581](https://issues.apache.org/jira/browse/REEF-581)

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Examples.DriverRestart/Org.Apache.REEF.Examples.DriverRestart.csproj [hunk 1]
```diff
@@ -17,11 +17,58 @@
...
+    </ProjectReference>
+    <ProjectReference Include="$(SolutionDir)\Org.Apache.REEF.Utilities\Org.Apache.REEF.Utilities.csproj">
+      <Project>{79e7f89a-1dfb-45e1-8d43-d71a954aeb98}</Project>
+      <Name>Org.Apache.REEF.Utilities</Name>
+    </ProjectReference>
+    <ProjectReference Include="$(SolutionDir)\Org.Apache.REEF.Wake\Org.Apache.REEF.Wake.csproj">
+      <Project>{cdfb3464-4041-42b1-9271-83af24cd5008}</Project>
+      <Name>Org.Apache.REEF.Wake</Name>
+    </ProjectReference>
+    <ProjectReference Include="..\Org.Apache.REEF.Examples\Org.Apache.REEF.Examples.csproj">
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Examples.DriverRestart/Org.Apache.REEF.Examples.DriverRestart.csproj]: Use `$(SolutionDir)` instead of `..`

**Comment by markusweimer** [comment]
@afchung Done with my pass. One minor comment.

**Comment by afchung** [comment]
@markusweimer Fixed. Please have another look. Thanks!

File: lang/cs/Org.Apache.REEF.Examples/DriverRestart/HelloRestartDriver.cs [hunk 2]
```diff
@@ -42,14 +43,16 @@ namespace Org.Apache.REEF.Examples.DriverRestart
     /// On restart, it expects all 5 of the running tasks to report back to it.
     /// </summary>
     public sealed class HelloRestartDriver : IObserver<IDriverRestartCompleted>, IObserver<IAllocatedEvaluator>, IObserver<IDriverStarted>,
```
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.Examples/DriverRestart/HelloRestartDriver.cs]: This class can be `internal`, right?

**Comment by markusweimer** [comment]
@afchung Found another nitpick. Also, I assume you tested this on HDI?

**Comment by afchung** [comment]
@markusweimer Fixed and ran the standard suite of tests on HDInsight. Thanks!

**Comment by markusweimer** [comment]
@afchung Excellent, will merge.

**Comment by markusweimer** [comment]
Turns out that the driver can't be `internal`. I'll fix that in the merge

**Pull request closed without merging**